x-pack/test/api_integration/config_security_basic.js

```diff
@@ -13,6 +13,7 @@ export default async function({ readConfigFile }) {
     config.esTestCluster.serverArgs = [
       'xpack.license.self_generated.type=basic',
       'xpack.security.enabled=true',
+      'xpack.security.authc.api_key.enabled=true',
```

Comment:
__note:__ it's too bad we don't have an easy way to reconfigure ES during tests and jest_integration tests are not available in xpack either. Ideally we'd like to check the case when keys are disabled so that we can catch the breaking changes in ES error format, but having a dedicated config just to test this is too much as well. No need to change/improve anything here, just grumbling about imperfect test system we live with.

Reply:
I agree! I almost created a dedicated test suite for the disabled case, but like you mentioned, it's not worth the effort/overhead